src/main/java/org/spongepowered/api/item/inventory/InventoryArchetype.java

```diff
@@ -130,6 +143,15 @@ default Builder title(Text title) {
         Builder property(InventoryProperty<String, ?> property);
 
         /**
+         * Sets the container provider. Use null to fallback to a basic container.
+         * TODO describe more / example? e.g. AnvilContainer can Repair basic container cannot.
+         *
+         * @param containerProvider the container provider
+         * @return Fluent pattern
+         */
+        Builder container(@Nullable BiFunction<Inventory, Player, Container> containerProvider);
```

Comment:
Could I ask about when the function could possibly change to provide more information (arguments?) I would rather a plain object interface to avoid having to royally break the method in the future (allows for future deprecation and such).